common-messaging/src/main/scala/org/genivi/sota/messaging/kinesis/KinesisClient.scala

```diff
@@ -51,32 +51,46 @@ object KinesisClient {
       val client = new AmazonKinesisClient(credentials, clientConfig)
       client.configureRegion(region)
       system.registerOnTermination(client.shutdown())
+      client
+    }
+
+  def getDeviceSeenPublisher(system: ActorSystem,
+                             config: Config): ConfigException Xor (DeviceSeenMessage => Unit) =
+    getAmazonClient(system, config).map { client =>
       (msg: DeviceSeenMessage) =>
         {
-          import io.circe.syntax._
-          import org.genivi.sota.marshalling.CirceInstances._
-          client.putRecord(streamName, ByteBuffer.wrap(msg.asJson.noSpaces.getBytes), msg.deviceId.underlying.get)
+          client.putRecord(DeviceSeenMessage.getClass.getName, ByteBuffer.wrap(msg.asJson.noSpaces.getBytes),
```

Comment:
If you are using a class as a partition key then you are actually prevents kinesis from sharding the messages